### ISSUE TITLE
[일지] 2026-09-02 진행 상황 · 09-03 예정 — 팀 4인

### ISSUE THREAD

**Issue by devlee328288** [devlee328288/Alpha_Stack#79]
4인 각자의 2026-09-02 진행 상황과 09-03 예정 사항을 한자리에 모읍니다.
서로 무엇을 하고 있는지, 어디서 맞물리는지 보이도록 관련 이슈를 함께 걸었습니다.

> 🙏 **팀원분들은 아래 자기 칸에 댓글로 남겨 주시면 본문에 옮기겠습니다.**
> 이전 판: #50 (09-01 → 09-02)

---

### 2026-09-02 진행 상황

**오늘의 큰 줄기는 "액면분할로 끊긴 가격을 잇고, 그 자료를 팀에 다시 내는 것"이었습니다.**

| 한 일 | 결과 |
|---|---|
| 🔴 수정주가 적재 (마이그레이션 v9) | `daily_price` 에 `adj_*` 4칸 + 출처 칸 · **9,223,644행 전부(100%)** 채움 |
| 실측 거래일 달력 표 신설 | `trading_calendar` 12,306행 · 조회 599ms → **13ms (45배)** |
| 🔴 재적재가 수정주가를 지우던 결함 수정 | `INSERT OR REPLACE` → `ON CONFLICT DO UPDATE` |
| **팀 자료 재반출 + HF 업로드** | 5,989,308 → **7,888,945행 (+31.7%)** · 검증 전량 통과 |
| 데이터셋 카드 전면 보강 | adj 사용법 · 한글 인코딩 · 개발구간 범위 |
| 테스트·문서 | 819 → **831 passed** · ERD v2.1 · TIL · 노트북 37셀 |

**무엇이 문제였나** — `close` 는 KRX 원문 그대로라 액면분할이 조정돼 있지 않습니다.
그대로 수익률을 내면 삼성전자 2018-05-04 이 **−98.04%** 로 읽힙니다(실제 −2.08%).
분할·병합이 **1,139건 · 806종(21.9%)** 에 걸쳐 있었습니다. (장환님 제보 → #51)

**막힌 곳 둘**

1. **FDR 이 최근 3,000거래일만 줍니다** (2014-06-13~). `count` 를 6000·9000 으로 올려도
   서버가 자르는 것을 실측했고, pykrx 도 같은 경로였습니다. 우리 자료의 23%가 비는데
   홀드아웃이 20240901 이라 **그 구멍은 전부 학습구간 안**이었습니다.
   → FDR 이 닿는 가장 이른 날을 앵커로 삼아 그 앞을 조정계수로 이어 붙였습니다.
   **대조할 것이 없는 구간이라**, FDR 을 일부러 앵커 하루만 남기고 2,999일을 우리 계산으로
   채워 진짜 값과 대조했습니다 — **오차 상한 최대 0.39%**.
2. **FDR 이 네 칸을 따로 반올림해 고가가 종가보다 낮아집니다.** 표본 3종에서 68행.
   `true_range`·`parkinson_20` 이 여기서 음수를 뱉습니다.
   → 배율 하나를 네 칸에 똑같이 곱해 전체에서 **위반 0행**.

**🔴 팀원분들께 — 수익률은 이제 `close` 가 아니라 `adj_close` 입니다**

| 무엇을 하려나 | 어느 칸 |
|---|---|
| 수익률 · 라벨 · 모멘텀 | **`adj_close`** |
| 변동성 (ATR · Parkinson · True Range) | **`adj_high`·`adj_low`·`adj_open`** |
| 시가총액 | `close` (`market_cap = close × listed_shares` 라서) |

- 거래정지일은 `adj_open`·`adj_high`·`adj_low` 가 **비어 있습니다**(체결이 없던 날).
  `0` 으로 채우면 "그 날 0원" 이 됩니다
- 지수(`index_price`)에는 `adj_*` 가 없습니다 — 계산값이라 분할 문제가 없습니다
- **한글이 깨져 보이면 파일이 아니라 읽는 방법입니다.** 이 PC 기본이 `cp949` 라
  `open(path, encoding="utf-8")` 이 필요합니다. HF 원격 바이트로 파일이 정상임을 확인했습니다

**남긴 것**

- 노트북에 **칸별 기본 통계**(결측률·기간·종목 수)를 더해 카드와 함께 다시 올리기
- `trading_calendar.next_session()` 이 아직 선형 탐색 — 이진탐색으로 바꿀 자리

### 2026-09-03 예정

- [ ] **HF 카드·노트북에 행 단위 기본 정보 보강 후 재업로드** — 칸별 결측률·기간·종목 수·분포
- [ ] 공공데이터포털 수집 경로 착수 (다음 수집원)
- [ ] `next_session()` 이진탐색 전환 (반입 검사가 행마다 부르는 자리)
- [ ] 팀원 PR 머지분 파트 문서화 (모델파트·피처파트 폴더)
- [ ] 열린 이슈 정리 — #44 · #45 · #52 (반입 규칙 3건)

> 관련: #51 (완료 · PR #78) · #76 (완료) · #67 · #29 · #63

---

### 2026-09-02 진행 상황

- QFRS 기반 백테스트 지표 공부
- `evaluation-features` 성과지표 구현
- 백테스트 지표 구현
- 5트랙 중첩 기획 및 설계

### 2026-09-03 예정

- [ ] 5트랙 중첩 구현
- [ ] `evaluation/metrics.py` 의 수수료·슬리피지 반영 등 지표 이후 작업을 참고해 평가지표 개선 방안 연구
- [ ] 비용 모델 4수준 기획 및 설계
- [ ] 기준선 3종 기획 및 설계
- [ ] ARIMA

> 참고: 수익률·성과지표를 `close` 로 계산하고 계셨다면 **`adj_close` 로 바꿔야 합니다.**
> `evaluation/horizon.py` 세 곳이 아직 원가격을 봅니다 — 제 파트 밖이라 손대지 않았습니다.
> 제가 고쳐도 될지, 민석님이 하실지 알려 주세요 (#51 에서 여쭀던 것과 같은 건입니다).

---

### 2026-09-02 진행 상황

**✅ 한 것**

1. **모델 조합 A~F 구성 및 기본 실험**
   - LogisticRegression · RandomForest · XGBoost · LightGBM 4종 구성
   - 모델별 기본 가중치와 `balanced` 성능 비교
   - 조합별 모델 비교표 · 설명 · 피처 선정 문서 작성

2. **수익률 피처 확장 — 전체 96개 모델 실험**
   - 기본 / `Daily_Return` / `5Day_Return` / `Daily_Return`+`5Day_Return` 구성
   - A~F 6조합 × 피처 구성 4개 × 모델 4종 = **노트북 96개**
   - Accuracy · Macro F1 · 하락 Recall 조화평균 기준 best result 정리

3. **홀드아웃 2년 변경 및 데이터 누출 방지** (#63)
   - `HOLDOUT_START` 를 `20240901` 로 변경
   - 모델 공통 데이터 필터와 경계 직전 5거래일 제거 적용
   - 테스트 **746 passed · 3 skipped · 3 xfailed**, 변경 범위 Ruff 통과
   - `feat/retrain-models-2y-holdout` 브랜치 커밋 · 푸시 완료

4. **Hugging Face 데이터 다운로드 및 전수 확인**
   - `qurious-quant/alphastack-krx-dev` 전체 12개 파일 다운로드
   - 데이터 파일 9개의 날짜 범위 및 MANIFEST SHA-256 검사
   - `full/daily_price_dev.parquet` — 5,989,308행 · 2010-01-04~2021-08-31
   - `full/index_price_dev.parquet` — 135,879행 · 2010-01-04~2021-08-31
   - 전체 파일에서 2021-09-01 이후 데이터 **0행** 확인
   - 2024년 개발구간 재반출 요청 이슈 #76 등록

### 2026-09-03 예정

- [ ] 2024-08 까지 포함된 새 개발구간 데이터 확보
- [ ] 새 개발구간에서 기본/`balanced` 가중치 재비교
- [ ] A~F 전체 96개 모델 재실행
- [ ] 새 결과로 조합별 best result 전면 갱신

> **재반출 완료했습니다** (#76). 새 개발구간은 **20100104 ~ 20240830 (14.7년)** 이고
> 이전보다 **+1,899,637행(+31.7%)** 입니다. `20240831` 은 **끝이지 시작이 아닙니다** —
> 96개 모델 재실행은 전 구간으로 잡아 주세요.

---

### 2026-09-02 진행 상황

- 몸 상태 이슈로 09-02 예정 작업을 **09-03 으로 미룸**

### 2026-09-03 예정

- [ ] 09-02 에 하려던 작업 수행
- [ ] 추가로 진행한 것이 있으면 09-03 팀 일정 이슈에 남길 예정

> 건강 먼저 챙기세요. 급한 건은 09-03 에 맞춰 두겠습니다.
>
> **제보하신 액면분할 건(#51)이 해결됐습니다.** 감사합니다 — 전수로 재 보니
> 1,139건 · 806종이었습니다. `close` 파생 피처 19개는 **`adj_close` 기준으로 재계산**이
> 필요합니다.

---

## 오늘 맞물린 곳

```
신장환 제보 (#51)  →  이동원 전수 실측  →  v9 적재 (PR #78)
                                              ↓
오준영 재반출 요청 (#76)  ──────────────→  HF 재배포 (+31.7% · adj 포함)
                                              ↓
                                  신장환 피처 19개 재계산 ·
                                  강민석 성과지표 · 오준영 96모델 재학습
```

관련: #51 · #76 · #67 · #63 · #33 · #29 · PR #78

**Comment by devlee328288** [comment]
@rkdalstjr 민석님, 오늘(09-03) 세션 시작 점검에서 `ruff check .` 가 **6건**을 냈는데 전부 PR #82 머지분 두 파일입니다. 제 파트 밖이라 손대지 않았습니다.

| 파일 | 건 |
|---|---|
| `evaluation/evaluation_backtest.py` | import 정렬(I001) · 파일 끝 개행 없음(W292) |
| `evaluation/evaluation_risk.py` | import 정렬(I001) · 자리표시 없는 f-string(F541) 등 |

4건은 자동으로 고쳐집니다.

```bash
ruff check evaluation/ --fix
```

다음 PR 에 함께 넣어 주시면 `ruff check .` 가 다시 0건이 됩니다. 급하지 않습니다.

**Comment by rkdalstjr** [comment]
네 PR에 적용해서 확인 부탁드립니다. 이슈를 어떻게 이용하는지 이제 감이 와서요..ㅎㅎ 이렇게 정보 주시고 있다는걸 몰랏습닞다
